.github/workflows/ci.yml

```diff
@@ -53,7 +53,7 @@ jobs:
           dotnet publish -f net7.0 -c Release ./test/test-applications/integrations/TestApplication.Smoke
           export OTEL_DOTNET_AUTO_HOME="${PWD}/bin/tracer-home"
           . ./instrument.sh
-          ./test/test-applications/integrations/TestApplication.Smoke/bin/Release/net7.0/publish/TestApplication.Smoke
+          ./test/test-applications/integrations/bin/TestApplication.Smoke/Release/net7.0/publish/TestApplication.Smoke
```

Comment:
Isn't it going to overwrite the same dir contents always? 

upstream was using `./test/test-applications/integrations/{Application}/bin/{Config}/{Framework}/{PackageVersion}/` pattern, this seems also more natural to me.

Reply:
I have to change default `obj` and `bin` folder location to be able to compile test application with multiple librariyversions.
The reason - dotnet creates `cs` files under the bin folder. When I have tried to use
`./test/test-applications/integrations/{Application}/bin/{Config}/{Framework}/{PackageVersion}/` for version 1 and version 2. 1st complilation was passing, but the second one was failing due to duplicates attribiutes in this `cs`
 files.

See https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/0182a430d13cd129a429daa4655319e3585b4ff6/test/test-applications/integrations/Directory.Build.props#L6-L9

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/2240